compressors/rANS.py

```diff
@@ -295,43 +283,19 @@ def decode_block(self, encoded_bitarray: BitArray):
             num_bits_consumed += num_symbol_bits
 
         # Finally, as a sanity check, ensure that the end state should be equal to the initial state
-        assert state == self.params.initial_state(self.freqs)
+        assert state == self.params.INITIAL_STATE
 
         return DataBlock(decoded_data_list), num_bits_consumed
 
 
 ######################################## TESTS ##########################################
 
 
-def _test_rANS_coding(freq, rans_params, data_size, seed):
-    """Core testing function for rANS"""
-    prob_dist = freq.get_prob_dist()
-
-    # generate random data
-    data_block = get_random_data_block(prob_dist, data_size, seed=seed)
-
-    # get optimal codelen
-    avg_log_prob = get_mean_log_prob(prob_dist, data_block)
-
-    # create encoder decoder
-    encoder = rANSEncoder(freq, rans_params)
-    decoder = rANSDecoder(freq, rans_params)
-
-    is_lossless, encode_len, _ = try_lossless_compression(
-        data_block, encoder, decoder, add_extra_bits_to_encoder_output=True
-    )
-
-    # avg codelen ignoring the bits used to signal num data elements
-    avg_codelen = encode_len / data_block.size
-    print(f"rANS coding: Optical codelen={avg_log_prob:.3f}, rANS codelen: {avg_codelen:.3f}")
-    assert is_lossless
-
-
 def test_check_encoded_bitarray():
```

Comment:
In the comment: 
`    # NOTE: the encoded_bitstream looks like = [<data_size_bits>, <final_state_bits>,<s0_bits>, <s1_bits>, ..., <s3_bits>]
`

Shouldn't it be:
`    # NOTE: the encoded_bitstream looks like = [<data_size_bits>, <final_state_bits>,<s_n-1_bits>, <s_n-2_bits>, ..., <s0_bits>]
`
I am assuming s_0 is the state you get after encoding first symbol. in this case after encoding A. You are prepending the states. Code makes sense, just trying to see if I understand consistently wrt comments.

Reply:
Thanks ll fix. Also will add details on alternative reverse approach

Reply:
done